formats/json-tests/jvmTest/src/kotlinx/serialization/BigDecimalTest.kt

```diff
@@ -0,0 +1,220 @@
+            ),
+        )
+
+        val encoded = json.encodeToString(bdHolder)
+
+        assertEquals(
+            """
+                {
+                    "bd": $bdExpected1,
+                    "bdList": [
+                        $bdExpected1
+                        $bdExpected2
+                        $bdExpected3
+                    ],
+                    "bdMap": {
+                        "$bdExpected1": $bdExpected2,
+                        "$bdExpected3": $bdExpected4,
+                        "$bdExpected5": $bdExpected6
+                    }
+                }
+            """.trimIndent(),
+            encoded,
+        )
+
+        val decoded = json.decodeFromString(BigDecimalHolder.serializer(), encoded)
+
+        assertEquals(
+            BigDecimal(bdExpected1),
+            decoded.bd,
+        )
+        assertEquals(
+            listOf(
+                BigDecimal(bdExpected1),
+                BigDecimal(bdExpected2),
+                BigDecimal(bdExpected3),
+            ),
+            decoded.bdList,
+        )
+        assertEquals(
+            mapOf(
+                BigDecimal(bdExpected1) to BigDecimal(bdExpected2),
+                BigDecimal(bdExpected3) to BigDecimal(bdExpected4),
+                BigDecimal(bdExpected5) to BigDecimal(bdExpected6),
+            ),
+            decoded.bdMap,
+        )
+    }
+
+    companion object {
+
+        // test data
+        private val bdActual1 = BigDecimal("725345854747326287606413621318.311864440287151714280387858224")
+        private val bdActual2 = BigDecimal("336052472523017262165484244513.836582112201211216526831524328")
+        private val bdActual3 = BigDecimal("211054843014778386028147282517.011200287614476453868782405400")
+        private val bdActual4 = BigDecimal("364751025728628060231208776573.207325218263752602211531367642")
+        private val bdActual5 = BigDecimal("508257556021513833656664177125.824502734715222686411316853148")
+        private val bdActual6 = BigDecimal("027134584027580606401102614002.366672301517071543257300444000")
+
+        private const val bdExpected1 = "725345854747326287606413621318.311864440287151714280387858224"
+        private const val bdExpected2 = "336052472523017262165484244513.836582112201211216526831524328"
+        private const val bdExpected3 = "211054843014778386028147282517.011200287614476453868782405400"
+        private const val bdExpected4 = "364751025728628060231208776573.207325218263752602211531367642"
+        private const val bdExpected5 = "508257556021513833656664177125.824502734715222686411316853148"
+        private const val bdExpected6 = "027134584027580606401102614002.366672301517071543257300444000"
+    }
+
+}
+
+@Serializable
+private data class BigDecimalHolder(
+    @Contextual
+    val bd: BigDecimal,
+    @Contextual
+    val bdList: List<@Contextual BigDecimal>,
+    val bdMap: Map<@Contextual BigDecimal, @Contextual BigDecimal>,
+)
```

Comment:
I think the repetition of `@Contextual` here would be reduced by https://github.com/Kotlin/kotlinx.serialization/issues/507. (Of course here this can be set with a file-annotation, but in other codebases it's likely that BigDecimal is used in many, many files.